include/SDL3/SDL_pen.h

```diff
@@ -0,0 +1,265 @@
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+/**
+ *  \file SDL_pen.h
+ *
+ *  Include file for SDL pen event handling.
+ *
+ *  This file describes operations for pressure-sensitive pen (stylus and/or eraser) handling, e.g., for input
+ *    and drawing tablets or suitably equipped mobile / tablet devices.
+ *
+ *  To get started with pens:
+ *  - Listen to ::SDL_PenMotionEvent and ::SDL_PenButtonEvent
+ *  - To avoid treating pen events as mouse events, ignore  ::SDL_MouseMotionEvent and ::SDL_MouseButtonEvent
+ *    whenever "which" == ::SDL_PEN_MOUSEID.
+ *
+ *  This header file describes advanced functionality that can be useful for managing user configuration
+ *    and understanding the capabilities of the attached pens.
+ *
+ *  We primarily identify pens by ::SDL_PenID.  The implementation makes a best effort to relate each :SDL_PenID
+ *    to the same physical device during a session.  Formerly valid ::SDL_PenID values remain valid
+ *    even if a device disappears.
+ *
+ *  For identifying pens across sessions, the API provides the type ::SDL_GUID .
+ */
+
+#ifndef SDL_pen_h_
+#define SDL_pen_h_
+
+#include "SDL_error.h"
+#include "SDL_guid.h"
+#include "SDL_stdinc.h"
+
+/* Set up for C function definitions, even when using C++ */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef Uint32 SDL_PenID; /**< SDL_PenIDs identify pens uniquely within a session */
+
+#define SDL_PEN_INVALID ((Uint32)0) /**< Reserved invalid ::SDL_PenID is valid */
+
+#define SDL_PEN_MOUSEID ((Uint32)-2) /**< Device ID for mouse events triggered by pen events */
+
+#define SDL_PEN_INFO_UNKNOWN (-1) /**< Marks unknown information when querying the pen */
+
+/**
+ * Pen axis indices
+ *
+ * Below are the valid indices to the "axis" array from ::SDL_PenMotionEvent and ::SDL_PenButtonEvent.
+ * The axis indices form a contiguous range of ints from 0 to ::SDL_PEN_AXIS_LAST, inclusive.
+ * All "axis[]" entries are normalised to either 0..1 (unidirectional axes) or between -1..1 (bidirectional axes).
+ * Unsupported entries are always "0.0f".
+ *
+ */
+typedef enum
+{
+    SDL_PEN_AXIS_PRESSURE = 0,               /**< Pen pressure.  Unidirectional: 0..1.0 */
+    SDL_PEN_AXIS_XTILT,                      /**< Pen horizontal tilt fraction.  Bidirectional: -1.0..1.0 (left-to-right).  This is NOT the angle, but the tilt vector x component.  The physical max/min tilt may be smaller than -1.0 / 1.0, cf. \link SDL_PenCapabilityInfo \endlink .  */
+    SDL_PEN_AXIS_YTILT,                      /**< Pen vertical tilt fraction.  Bidirectional: -1.0..1.0 (top-to-bottom).  This is NOT the angle, but the tilt vector y component.  */
+    SDL_PEN_AXIS_DISTANCE,                   /**< Pen distance to drawing surface.  Unidirectional: 0.0..1.0 */
+    SDL_PEN_AXIS_ROTATION,                   /**< Pen barrel rotation.  Bidirectional: -1.0..1.0 (clockwise, 0 is facing up).  Multiply with "2.0 * M_PI" to get radians, or "360.0" for degrees. */
```

Comment:
This doesn't seem right. Multiplying -1.0 by 360 would give you a -360 degree rotation.

Reply:
Thanks for catching this!  Will update the documentation (for now).
I went with -1.0..1.0 here, but this is not a straightforward call.  My rationale is below:

- Should we go from -x/2..x/2 or from 0..x?
- Should x be 1.0, π, or 360?

## Negative-Positive or 0-Positive?
- negative-positive: Going from e.g. -1.0 to 1.0 with 0.0 at the centre means that "normal" pen usage will report 0, while barrel-rotating left will return negative values and right will return positive values.  This symmetry makes it easy to e.g. set thresholds, but it means that rotating the pen by 180 degrees is ambiguous (is it -1.0 or 1.0?).  Thus, the symmetry breaks at this point.
- 0..positive: Going from e.g. 0.0 to 1.0 (without reaching 1.0) avoids the problem with 180 degree rotation, but setting a threshold for e.g. "less than 45 degrees left" now becomes "rotation > 0.875".

## Range
- Unit range (0..1) or -1..1: Homogenous wrt the other axis values, easy to convert to radians or degrees, but not a common format.
- radians (-π..π or 0..2π): Probably the most common format, and likely the most directly useful; I expect that most uses of the angle will pass it directly into `sin()` and `cos()` anyway.  However, IEEE-754 can't represent π and its multiples/fractions precisely, so anything that wants to look for exact right angles will be annoying.
- degrees (0..359 or -180..179): This is what `SDL_RenderTextureRotated()` takes as parameters, and it is what Win32 and Wayland report (though not XInput2).  May help port code, probably the most intuitive way for writing thresholds in code, but requires conversion to radians for most other use cases.

I'd personally find radians most natural, but nobody else uses them in SDL (or in other existing event APIs), so let's rule them out.  The choice then is between being most like what people would expect for a feature that describes rotation (0..359, apparently?), and what people would most expect the range of values in the axis array to look like (0.0..1.0 or -1.0..1.0).  Since the latter is also more straightforward to convert to radians, I went for that.